dspace-server-webapp/src/main/java/org/dspace/app/rest/authorization/impl/ManageGroupFeature.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.authorization.impl;
+
+import java.sql.SQLException;
+
+import org.dspace.app.rest.authorization.AuthorizationFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeatureDocumentation;
+import org.dspace.app.rest.model.BaseObjectRest;
+import org.dspace.app.rest.model.GroupRest;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.eperson.Group;
+import org.dspace.eperson.service.GroupService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * The Manage Group Feature. It can be used to verify if the current user can manage a specific group.
+ *
+ * Authorization is granted if the current user has site-wide ADMIN permissions or if the current user
+ * is a member of this group in case the group is linked to a community or collection.
+ */
+@Component
+@AuthorizationFeatureDocumentation(name = ManageGroupFeature.NAME, description =
+    "It can be used to verify if the current user can manage a specific group")
+public class ManageGroupFeature implements AuthorizationFeature {
+
+    public final static String NAME = "canManageGroup";
+
+    @Autowired
+    GroupService groupService;
+
+    @Autowired
+    AuthorizeService authorizeService;
+
+    @Override
+    public boolean isAuthorized(Context context, BaseObjectRest object) throws SQLException {
+        // this authorization only applies to GroupRest
+        if (!(object instanceof GroupRest)) {
+            return false;
+        }
+        GroupRest groupRest = (GroupRest) object;
+
+        // get the group
+        Group group = groupService.findByName(context, groupRest.getName());
+        if (group == null) {
+            return false;
+        }
+
+        // admins can manage any group
+        if (authorizeService.isAdmin(context)) {
+            return true;
+        }
+
+        // community/collection admins cannot manage special groups
+        if (
+            group.equals(groupService.findByName(context, Group.ANONYMOUS)) ||
+            group.equals(groupService.findByName(context, Group.ADMIN))
+        ) {
+            return false;
+        }
```

Comment:
why this specific check? the `authorizeService` should already deal with it if needed. If we really need to make a if statement here better to compare the group name with the constants as we already have it and this would avoid any database query

Reply:
This was added because `org.dspace.authorize.AuthorizeServiceImpl#isAdmin(Context c, EPerson e, DSpaceObject o)` returns true for Anonymous group for both Community/Collection admins and this feature is so that only collection/community related groups (of which they are admin) return true. 
`isAdmin` returns true because the parent result [here](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authorize/AuthorizeServiceImpl.java#L405) of the Anonymous group is the collection, because the collection has `DEFAULT_ITEM_READ` Anonymous RP, checked [here](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java#L736-L739).
(tested with IT `collection1AdminCanNotManageAnonymousGroup`/`community1AdminCanNotManageAnonymousGroup`)

I'm assuming that it is not general 'issue' that a com/col admin is considered admin of the groups of their `DEFAULT_ITEM_READ` Anonymous RP? So we can just keep the Anon group check (Admin isn't needed) but on name instead of with `groupService.findByName` or just allow `canManageGroup=true` check on Anonymous group to also return true (is not editable in Angular anyway since it's a permanent group, this would mean the comcol admin could access the group page and add/delete groups/epersons but this doesn't actually affect Anonymous group, since it is everyone)

Reply:
@MarieVerdonck : In my opinion (though I'd like to hear from @abollini and @benbosman) it seems like you may have found a bug in how `org.dspace.authorize.AuthorizeServiceImpl#isAdmin(Context c, EPerson e, DSpaceObject o)` functions with the Anonymous group.  I don't think that method should return `true` for Community/Collection Admins when the Anonymous group is passed as the object.

We might consider moving your `if` logic (with a brief comment) into that `AuthorizeServiceImpl#isAdmin()` method directly (as that'd ensure it blocks non-Admins from managing the Anonymous group throughout the system).  However, if we have concerns about doing that low level change (or it results in side effects in other tests), then we may want to keep your PR as-is and log this issue as a separate bug ticket so that we can spend more time investigating how best to fix it.

Reply:
I agree with @tdonohue here you have spot a bug that could be also very dangerous as a default item read permission can eventually turn in permission to manage a global group, the anonymous in your test but eventually also the administrators.

The key should be this comment https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java#L714 and the relevant check https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java#L722 if we introduce the same also in the policy section we should have done

Reply:
@MarieVerdonck or @benbosman : Any thoughts on the last two comments from me & @abollini ?  

If you feel this bug is complex to resolve quickly & safely, I'm OK with creating a separate ticket for the bug in  `org.dspace.authorize.AuthorizeServiceImpl#isAdmin(Context c, EPerson e, DSpaceObject o)`, and ensuring it gets resolved immediately after the 7.0 release (if not before).  So, please let me know what you think.

Reply:
This can indeed give a community/collection admin permission to manage any group that is put in its resource policies for `Constants.DEFAULT_BITSTREAM_READ` or `Constants.DEFAULT_ITEM_READ`. Not sure if there is already a way to change either of these to administrators group for example, like there was for DSpace6 under 'Administer Authorization Policies' (`{url}/admin/authorize`), but this implies there is a way to alter which groups are on those RP.
But a com/col admin being able to manage their ItemRead/BitstreamRead group was added deliberately, as evidenced by IT `GroupRestRepositoryIT#commAdminAndColAdminCanManageBitstreamReadGroupTest`, so that would have to be changed as well at least if you remove this code.
@abollini Not sure how a size check on policies [here](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java#L736) would prevent this? 
In this case there is just one ItemRead RP on Anon group, parent of RP is the collection of which user is admin => can manage Anon group.

Reply:
> This can indeed give a community/collection admin permission to manage any group that is put in its resource policies for `Constants.DEFAULT_BITSTREAM_READ` or `Constants.DEFAULT_ITEM_READ`. 

@MarieVerdonck : Thanks for that explanation, I had previously misunderstood the main reason for this behavior, but that statement now clarifies things in my mind.

It now makes sense to me that currently this behavior is necessary to allow the Coll/Comm Admin to manage the `DEFAULT_BITSTREAM_READ` and/or `DEFAULT_ITEM_READ` permissions. 

Longer term (not in this PR or even in 7.0), I still think that current behavior sounds slightly incorrect, so I went ahead and created a separate ticket for this: #3323 (Please feel free to enhance or add to it, if anything I said is incorrect)

That'd mean that, for the purposes of this PR, I think it's fine to leave this as-is, but we should add an inline comment to explain why this is needed.  Something like:
```
// Community/Collection admins cannot manage special groups
// This check is required until https://github.com/DSpace/DSpace/issues/3323 is fixed.
```

@abollini , for the purposes of this PR, I'd like to move this forward as-is, since the bug already exists separately (and has tests to prove the behavior).  So, as noted above, I've moved this to a separate ticket #3323 as we may need to solve this in 7.1

Reply:
Code updated with comment and to use the group name as to avoid using a DB query as @abollini suggested. 
This prevents at least Anon and Admin group to be managed in the above described way (putting it as group on itemRead/bitstreamRead RP), but not yet any other group. 
Alternatively the check here could be instead that if the group does not start with either `COLLECTION_` or `COMMUNITY_` to return false.

Reply:
@MarieVerdonck : I'd be in favor of adding that additional check, as it provides better security.  I do feel that Coll/Comm admins should be limited to managing `COLLECTION_*` or `COMMUNITY_*` groups.  It's obvious that our underlying Java API doesn't enforce that quite correctly yet, but this would be a good opportunity to enforce it at the REST API layer until we can investigate a better fix at the Java API layer.